resources/services/lightsail/instances.go

```diff
@@ -0,0 +1,455 @@
+						Type:        schema.TypeStringArray,
+					},
+					{
+						Name:        "cidrs",
+						Description: "The IPv4 address, or range of IPv4 addresses (in CIDR notation) that are allowed to connect to an instance through the ports, and the protocol",
+						Type:        schema.TypeStringArray,
+					},
+					{
+						Name:        "common_name",
+						Description: "The common name of the port information.",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "from_port",
+						Description: "The first port in a range of open ports on an instance",
+						Type:        schema.TypeInt,
+					},
+					{
+						Name:        "ipv6_cidrs",
+						Description: "The IPv6 address, or range of IPv6 addresses (in CIDR notation) that are allowed to connect to an instance through the ports, and the protocol",
+						Type:        schema.TypeStringArray,
+					},
+					{
+						Name:        "protocol",
+						Description: "The IP protocol name",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "to_port",
+						Description: "The last port in a range of open ports on an instance",
+						Type:        schema.TypeInt,
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchLightsailInstances(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	c := meta.(*client.Client)
+	svc := c.Services().Lightsail
+	input := lightsail.GetInstancesInput{}
+	for {
+		output, err := svc.GetInstances(ctx, &input, func(o *lightsail.Options) {
+			o.Region = c.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		res <- output.Instances
+
+		if aws.ToString(output.NextPageToken) == "" {
+			break
+		}
+		input.PageToken = output.NextPageToken
+	}
+	return nil
+}
+
+func fetchLightsailInstanceAddOns(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	instance := parent.Item.(types.Instance)
+	res <- instance.AddOns
+	return nil
+}
+
+func fetchLightsailInstanceHardwareDisks(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	instance := parent.Item.(types.Instance)
+	res <- instance.Hardware.Disks
+	return nil
+}
+
+func fetchLightsailInstanceHardwareDiskAddOns(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	disk := parent.Item.(types.Disk)
+	res <- disk.AddOns
+	return nil
+}
+
+func fetchLightsailInstanceNetworkingPorts(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	instance := parent.Item.(types.Instance)
+	res <- instance.Networking.Ports
```

Comment:
What if `Networking` is nil?

Reply:
Good catch! I looked at some other examples and the SDK code, and it looks like the right way to handle this is to return early without writing to the channel. I've updated the code to do this 👍